src/main/java/com/pusher/client/connection/websocket/WebSocketClientWrapper.java

```diff
@@ -51,27 +52,27 @@ public WebSocketClientWrapper(final URI uri, final WebSocketListener proxy) thro
                 throw new SSLException(e);
             }
         }
-
-        this.proxy = proxy;
+        this.webSocketListener = webSocketListener;
+        setProxy(proxy);
```

Comment:
My main thought was that we're calling out of code we control into a 3rd party library and it _may_ be worth avoiding that. It was just a thought.